### PR TITLE
Support for packaging as iOS frameworks

### PR DESCRIPTION
**Pull request by Jeehut**
In order to get this project working in a tvOS project that I'm working on right now I needed to update/change a few things:
- [x] package as Framework (for easier inclusion in iOS & tvOS) – #283
- [x] added Cocoapods tvOS support – #369
- [x] added setBundle: method for imageNamed: method (to make it work with frameworks including assets)

I'm currently testing the changes in the project and haven't started tvOS integration yet, so there might still be some glitches. But since it's basically working for me now I felt like making the pull request already. I'll keep you updated on the progress.

_UPDATE:_
After input from @adamgit this pull request is not intended to get tvOS working completely. This is for making framework packaging possible only (with some minor tvOS related changes included already).

### CODE REVIEW

**Comment by Jeehut** [comment]
This pull request is supposed to fix https://github.com/SVGKit/SVGKit/issues/369 and https://github.com/SVGKit/SVGKit/issues/283.

**Comment by adamgit** [comment]
We tried repackaging main project as a framework a while back, and we 
failed. Couldn't get it to work.

If you managed to get that working, is it possible to submit as a separate 
pull request? Because that's a huge change (big improvement, IMHO!) all on 
its own.

Then submit the tvos/podspec changes as an immediate followup.

there are more people who can test the framework bit than can test the tvos 
stuff, so.I'd like to get someone reviewing it first asap.

**Comment by Jeehut** [comment]
@adamgit The tvOS bit doesn't prevent anybody from testing the iOS framework bit and it didn't include any changes except marking the podspec as tvOS compatible (what won't do any harm) and creating a tvOS target alongside an iOS target. Therefore I don't think it's necessary to keep these things apart from each other.

I agree though, that we should make sure the framework integration works first. So, I have just updated the name of this pull request accordingly and removed the tvOS-related TODOs. I think we can keep the tvOS target and the one line within the podspec, don't you agree? I haven't done any other tvOS-related changes yet and will send those with a different pull request then if needed.

If you think we should remove the minor tvOS-related changes, let me know.

**Comment by Jeehut** [comment]
Btw: I say things are working for me, but what I mean is: Everything is building. I have done more changes on my project than only including this project as a framework and it's not all working yet. But I can't tell if it's an issue with this framework or the project, so others will need to test the framework integration in their projects, too.

**Comment by adamgit** [comment]
Yep, exactly, I'd like to get a few different heavy users testing this with 
their different set ups

File: Source/Sources/SVGKSourceLocalFile.m [hunk 1]
```diff
@@ -52,7 +60,14 @@ +(SVGKSourceLocalFile*) internalSourceAnywhereInBundleUsingName:(NSString*) name
 	
 	/** First, try to find it in the project BUNDLE (this was HARD CODED at compile time; can never be changed!) */
 	NSString *pathToFileInBundle = nil;
-	NSBundle *bundle = [NSBundle mainBundle];
+
```
Comment (adamgit) [inline Source/Sources/SVGKSourceLocalFile.m:41]: make this  a parameter to the method, rather than adding global state

Comment (Jeehut) [inline Source/Sources/SVGKSourceLocalFile.m:41]: Just did in 36314710829fce676c2d1b7464d6dec32ec76e27.

**Comment by adamgit** [comment]
@ ... everyone:

Please try this out, test it with your projects, and report back ASAP if any problems / crashes / failures. If it breaks bulding for anyone, I'll revert.

**Comment by adamgit** [comment]
@Dschee - what are the install / usage steps for people wanting to use this directly as framework?

I'm guessing:
1. Open SVGKit-iOS project (the original one)
2. Build
3. ... there's a Framework output somewhere?

Problem is ... I'm not seeing it

**Comment by Jeehut** [comment]
@adamgit: I'm not using this framework anymore but what I basically did back then when I was using SVGKit was to just drag and drop the `SVGKit-iOS.xcodeproj` to my project navigator, then I could choose `SVGKit.framework` within `Linked Frameworks and Libraries` in my targets `General` tab. Does this help?

**Pull request merged**